src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -41,14 +40,16 @@ public TypeFactory(OutputLibrary library, Type unknownType)
         public CSharpType CreateType(InputType inputType) => inputType switch
         {
             InputLiteralType literalType => CSharpType.FromLiteral(CreateType(literalType.LiteralValueType), literalType.Value),
-            InputUnionType unionType => CSharpType.FromUnion(unionType.UnionItemTypes.Select(CreateType).ToArray(), unionType.IsNullable),
+            InputUnionType unionType => CSharpType.FromUnion(unionType.UnionItemTypes.Select(x => CreateType(x)).ToArray(), unionType.IsNullable),
```

Comment:
looks like we do not need to change this line now

Reply:
yes, after removing format